src/SFA.DAS.AssessorService.Data/CertificateRepository.cs

```diff
@@ -38,9 +38,49 @@ public async Task<Certificate> New(Certificate certificate)
         {
             // Another check closer to INSERT that there isn't already a cert for this uln / std code
             var existingCert = await _context.Certificates.FirstOrDefaultAsync(c =>
-                c.Uln == certificate.Uln && c.StandardCode == certificate.StandardCode && c.CreateDay == certificate.CreateDay);
+                c.Uln == certificate.Uln && c.StandardCode == certificate.StandardCode &&
+                c.CreateDay == certificate.CreateDay);
             if (existingCert != null) return existingCert;
-            
+
+            _context.Certificates.Add(certificate);
+            try
+            {
+                _context.SaveChanges();
+            }
+            catch (Exception e)
+            {
+                if (!(e.InnerException is SqlException sqlException)) throw;
+
+                if (sqlException.Number == 2601 || sqlException.Number == 2627)
```

Comment:
I'm really not a fan of exception handling as flow, but I understand the pressures and don't want to drag.  Can a tech debt story be raised to precheck what causes these 2 error codes and we finesse out the exception as program stuff here?  I suggest a short spike story to investigate and establish under tech debt, put straight into the backlog for a less time-constrained point in the future.

Reply:
These are specifically checked for Unique Index Constraint violations as even with some very close checking for existence before doing an insert, we were still getting the odd ninja getting through.

Reply:
If its there to be belt when there's already braces applied, then it's fine as is, nothing else to do.  (I hope I'm not straining the metaphor too much there...)

Reply:
(though I'd maybe refactor out into a local method called LastStandMeasureToStopNinjaI|nsertViolationWeAreAlreadyGuardingAgainst...)